frontend/packages/console-shared/src/utils/label-filter.ts

```diff
@@ -6,5 +6,7 @@ import { requirementToString } from '@console/internal/module/k8s/selector-requi
 export const getLabelsAsString = (obj: K8sResourceCommon): string[] => {
   const { labels } = obj.metadata;
   const requirements = toRequirements(labels);
-  return _.map(requirements, requirementToString);
+  const stringRequirements = _.map(requirements, requirementToString);
+  // Remove trailing equals if any
+  return _.map(stringRequirements, (req) => req.replace(/=$/, ''));
```

Comment:
It looks like there is a bug in `toRequirements` where it should generate `=''`. (I believe this is the correct syntax, although it's unclear to me reading https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)

I would really like to treat these strings as black boxes because if we change `toRequirements` (e.g. to fix the problem above), this regex would break. Does the trailing `=` break filtering?

The real fix is not to convert these labels into strings at all IMO. We should only do that for display in the dropdown.

Reply:
With the trailing '=' things breakdown when working with Search Page. So if I pass `key` to search page it runs fine but if I pass `key=` it breaks down.

Reply:
` oc get nodes -l node-role.kubernetes.io/worker`
` oc get nodes -l node-role.kubernetes.io/worke=`
` oc get nodes -l node-role.kubernetes.io/worker=''`
All three variation work but for some reasons  `node-role.kubernetes.io/worker=` does not work with search. And console when showing labels with empty values does not show `=` sign.

Reply:
I think the root of the problem is we are using the same query parameter for both the label filtering. From `Search` and `ListPage` separating the query parameter should allow them to function independently.
It does not break filtering, but it was having some issues with Search Page.